.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,24 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
```

Comment:
Documentation on how to use this file would be useful. Especially for anyone who needs to maintain the dockerfile - I'm sure I will need to eventually, and may need to run this devcontainer instead of throwing changes at CI until they work).

Reply:
That's right. I thought that this devcontainer should be always as minimal as possible to replicate CI environment, but I've added a comment to clarify that. Only nuance is that entrypoint.sh or cmd is not run by default, which is better IMO as we allocate entrypoint.sh to the function of doing the actual CI check, not desirable during every container creation for booting a dev environment.

I totally agree that doing any tooling change within devcontainer is quite helpful as it shows a nearly-deterministic result, and also comfortably opens the environment without needing to maintain multiple versions of packages in the actual host system.

Reply:
All makes sense. How do you actually use this though, as a user? It seems you're supposed to launch the devcontainer and connect vscode to it. How do you do that? Is there an example tutorial somewhere we can link to?

Reply:
@kainino0x That's a really good suggestion, I've put two links, one for GitHub browser case and other for local VS Code! Eventually, we can mention this tooling in "Contributing" related sections as it really simplifies the flow, but I guess it is first better to test for a while and then extend into such visible spots. I'm having good time with it but didn't challenge it **very** hard